metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -108,9 +108,15 @@ def __init__(self, attributes=None, statically_defined=False):
             self.attributes["node_selector"] = self.parse_node_selector(self.attributes["node_selector"].split(","))
 
         if self.attributes["tolerations"]:
-            from kubernetes.client import V1Toleration
+            attribute_map = [
+                "effect",
+                "key",
+                "operator",
+                "toleration_seconds",
+                "value",
+            ]
             for toleration in self.attributes["tolerations"]:
-                invalid_keys = [k for k in toleration.keys() if k not in V1Toleration.attribute_map.keys()]
+                invalid_keys = [k for k in toleration.keys() if k not in attribute_map]
```

Comment:
Why did this have to change? As K8s changes, these attributes could change, right?

Reply:
When it runs on Argo, the module `kubernetes` is unavailable.
I can make the validation optional, given that it is required only when the flow runs locally and `KubernetesClient` raises an exception if the module is not installed. WDYT?
```python
if self.attributes["tolerations"]:
    try:
        from kubernetes.client import V1Toleration
        for toleration in self.attributes["tolerations"]:
            invalid_keys = [k for k in toleration.keys() if k not in V1Toleration.attribute_map.keys()]
            if len(invalid_keys) > 0:
                raise KubernetesException(
                    "Tolerations parameter contains invalid keys: %s" % invalid_keys
                )
    except (NameError, ImportError):
        pass
```

Reply:
The above looks like a happy compromise to me in which we try to do our best to validate AND keep up with upstream changes in K8s.